com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -227,9 +174,141 @@ public CommandLine GetShaderCommandLineOptions(String[] args) {
         return cmd;
     }
 
-    public void BuildShader(String[] args, ES2ToES3Converter.ShaderType shaderType, CommandLine cmd, IShaderCompiler shaderCompiler) throws IOException, CompileExceptionError {
+    public Platform getPlatformFromCommandLine(String[] args) throws CompileExceptionError {
+        CommandLine cmd = getShaderCommandLineOptions(args);
+        String platformName = cmd.getOptionValue("platform", "");
+        Platform platform = Platform.get(platformName);
+        if (platform == null) {
+            throw new CompileExceptionError(String.format("Invalid platform '%s'\n", platformName));
+        }
+        return platform;
+    }
```

Comment:
I think we should remove this and just pass in the platform as an argument instead..

Reply:
yes, it looks a bit... hmm. strange

Reply:
It’s legacy code with some intention not necessary any longer afaik